argocd/features.go

```diff
@@ -43,7 +43,7 @@ type ServerInterface struct {
 	ServerVersionMessage *version.VersionMessage
 	ProviderData         *schema.ResourceData
 
-	InitLock    sync.Mutex
+	InitLock    sync.RWMutex
```

Comment:
This is not enough, the locking mechanism has to be updated as well in the function with RLock/Lock and RUnlock/Unlock